tests/numeric/test_BaseNumericTransformer.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+import tests.test_data as d
+from tests.base_tests import GenericFitTests, GenericInitTests, GenericTransformTests
+
+
+class BaseNumericTransformerInitTests(GenericInitTests):
+    """
+    Tests for BaseNumericTransformer.init().
+    Note this deliberately avoids starting with "Tests" so that the tests are not run on import.
+    """
+
+
+class BaseNumericTransformerFitTests(GenericFitTests):
+    """
+    Tests for BaseNumericTransformer.fit().
+    Note this deliberately avoids starting with "Tests" so that the tests are not run on import.
+    """
+
+    def test_non_numeric_exception_raised(self, initialized_transformers):
+        """Test an exception is raised if self.columns are non-numeric in X."""
+        df = d.create_df_2()
+
+        x = initialized_transformers[self.transformer_name]
+
+        with pytest.raises(
+            TypeError,
+            match=rf"{self.transformer_name}: The following columns are not numeric in X; \['b'\]",
+        ):
+            x.fit(df, df["c"])
+
+
+class BaseNumericTransformerTransformTests(
+    GenericTransformTests,
+):
+    """
+    Tests for the transform method on BaseNumericTransformer.
+    Note this deliberately avoids starting with "Tests" so that the tests are not run on import.
+    """
+
+    def test_non_numeric_exception_raised(self, initialized_transformers):
+        """Test an exception is raised if self.columns are non-numeric in X."""
+        df = d.create_df_2()
+
+        x = initialized_transformers[self.transformer_name]
+
+        with pytest.raises(
+            TypeError,
+            match=rf"{self.transformer_name}: The following columns are not numeric in X; \['b'\]",
+        ):
+            x.transform(df)
+
+
+class TestInit(BaseNumericTransformerInitTests):
+    """Tests for BaseNumericTransformer.init()"""
+
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "BaseNumericTransformer"
+
+
+class TestFit(BaseNumericTransformerFitTests):
+    @classmethod
```

Comment:
N: Do we need a docstring along the lines of """Tests for BaseNumericTransformer.init()""" but for fit and transform?

Reply:
added here! https://github.com/lvgig/tubular/pull/266/commits

Reply:
Thanks!